src/formatDate.js

```diff
@@ -7,8 +7,54 @@
  *
  * @returns {string}
  */
+
 function formatDate(date, fromFormat, toFormat) {
-  // write code here
+  const parts = date.split(fromFormat[fromFormat.length - 1]);
+
+  const dateParts = {
+    DD: parts[fromFormat.indexOf('DD')],
+    MM: parts[fromFormat.indexOf('MM')],
+    YYYY: parts[fromFormat.indexOf('YYYY')],
+    YY: parts[fromFormat.indexOf('YY')],
+  };
+
+  const resultParts = [];
+
+  for (const format of toFormat) {
+    if (format === 'YYYY') {
+      resultParts.push(
+        formatYear(dateParts['YYYY'] || dateParts['YY'], format),
+      );
+    } else if (format === 'YY') {
+      resultParts.push(
+        formatYear(dateParts['YY'] || dateParts['YYYY'], format),
+      );
+    } else if (format === 'MM') {
+      resultParts.push(dateParts['MM']);
+    } else if (format === 'DD') {
+      resultParts.push(dateParts['DD']);
+    }
+  }
+
+  return resultParts.join(toFormat[toFormat.length - 1]);
```

Comment:
Similar to line 12, assuming the last element of `toFormat` is the delimiter may not be correct. Ensure that the delimiter is correctly identified or specified.